### DISCUSSION TITLE
Identifying significance for each level of a factor with envfit

### DISCUSSION THREAD

**Discussion by hkaspersen** [vegandevs/vegan#628] [Q&A]
Hello!
I am using `envfit` to see how two factor variables fit to an ordination.
I see in different tutorials that with numerical variables you get one p-value per variable. 
For the factor variables you only get one p-value per variable, but does this then mean that all the factor levels are significant?
I'm trying to only plot significant variables, but I'm a bit unsure how to interpret this, as I will get one arrow per factor level.
Any help is greatly appreciated!

**Comment by jarioksa** [comment]
You shouldn't get arrows per factor levels, but you should get centroids of factor levels. Function `envfit` has no way of assessing "significances" of single factor levels, but you really should get only a permutation _p_-value for the whole factor with all its levels. You should define your factor as an **R** `factor` for correct analysis. 

Actually, I am not quite sure how would you define significance of a single factor level. Somehow it would be a contrast of each single level against all other levels, and I'm not sure what would that mean.

**Reply by hkaspersen** [reply]
Thanks for the reply! 
I am wondering though, as I do get significance for each factor level in the `anova.cca` on the overall model with `capscale`. Is this similar, or completely different? Sorry I do not have a statistical background so all this is new to me!

**Reply by jarioksa** [reply]
The only way you get a _p_-value in `capscale` (or `adonis2` or any constrained ordination model) is with `by = "onedf"` and this gives you contrasts to the first level of the factor. This hardly makes sense in `envfit` and also will omit the first level of the factor.
